build: run four nodes in acceptance tests

tbg opened the pull request:
This is required for

https://github.com/cockroachdb/cockroach/pull/17272

and merits a separate review.

benesch commented on Makefile:
```diff
@@ -268,8 +269,9 @@ upload-coverage: $(BOOTSTRAP_TARGET)
 	@build/upload-coverage.sh
 
 .PHONY: acceptance
+acceptance: TESTTIMEOUT := $(ACCEPTANCETIMEOUT)
```
`export TESTTIMEOUT := $(ACCEPTANCETIMEOUT)` should do the same without the explicit `env` below!

tschottdorf replied:
Done.

a-robinson commented on pkg/acceptance/run.sh:
```diff
@@ -10,4 +10,4 @@ set -euxo pipefail
 export TMPDIR=$PWD/artifacts/acceptance
 
-make test PKG=./pkg/acceptance TAGS=acceptance TESTFLAGS="${TESTFLAGS--v -nodes 3} -l $TMPDIR"
+make test PKG=./pkg/acceptance TESTTIMEOUT="${TESTTIMEOUT-30m}" TAGS=acceptance TESTFLAGS="${TESTFLAGS--v -nodes 4} -l $TMPDIR"
```
Shouldn't this be `${TESTTIMEOUT:-30m}` (with the `:`)?

tschottdorf replied:
If you care about the case in which the variable is set but null? Anecdotally we're using `-` over `:-` in most places.

tbg committed:
build: run four nodes in acceptance tests

This is required for

https://github.com/cockroachdb/cockroach/pull/17272

and merits a separate review.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=17324) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=17324) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cockroachdb/cockroach?pullRequest=17324) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17324)
<!-- Reviewable:end -->
